src/site/content/en/blog/handwriting-recognition/index.md

```diff
@@ -0,0 +1,426 @@
+    });
+  }
+  startStroke(event);
+});
+```
+
+{% Aside 'caution' %} The current implementation on Chrome&nbsp;OS does not support grapheme sets
+yet, they are silently ignored. {% endAside %}
+
+#### Add a stroke
+
+The `pointerdown` event is also the right place to start a new stroke. To do so, create a new
+instance of `HandwritingStroke`. Also, you should store the current time as a point of reference for
+the subsequent points added to it:
+
+```js
+function startStroke(event) {
+  activeStroke = {
+    stroke: new HandwritingStroke(),
+    startTime: Date.now(),
+  };
+  addPoint(event);
+}
+```
+
+#### Add a point
+
+After creating the stroke, you should directly add the first point to it. As you will add more
+points later on, it makes sense to implement the point creation logic in a separate method. In the
+following example, the `addPoint()` method calculates the elapsed time from the reference timestamp.
+The temporal information is optional, but can improve recognition quality. Then, it reads the X
+and Y coordinates from the pointer event and adds the point to the current stroke.
+
+```js
+function addPoint(event) {
+  const timeElapsed = Date.now() - activeStroke.startTime;
+  activeStroke.stroke.addPoint({
+    x: event.offsetX,
+    y: event.offsetY,
+    t: timeElapsed,
+  });
+}
+```
+
+The `pointermove` event handler is called when the pointer is moved across the screen. Those points
+need to be added to the stroke as well. The event can also be raised if the pointer is not in a
+"down" state, for example when moving the cursor across the screen without pressing the mouse
+button. The event handler from the following example checks if an active stroke exists, and adds the
+new point to it.
+
+```js
+canvas.addEventListener('pointermove', (event) => {
+  if (activeStroke) {
+    addPoint(event);
+  }
+});
+```
+
+#### Recognize text
+
+When the user lifts the pointer again, you can add the stroke to your drawing by calling its
+`addStroke()` method. The following example also resets the `activeStroke`, so the `pointermove`
+handler will not add points to the completed stroke.
+
+{% Aside %} If necessary, you can also use the drawing's `getStrokes()` method to list all strokes,
+and the `removeStroke()` method to remove a particular one from the drawing. {% endAside %}
+
+Next, it's time for recognizing the user's input by calling the `getPrediction()` method on the
+drawing. Recognition usually only takes less than a few hundred milliseconds, so you can repeatedly run predictions
+if needed. The following example runs a new prediction after each completed stroke.
+
+```js
+canvas.addEventListener('pointerup', async (event) => {
+  drawing.addStroke(activeStroke.stroke);
+  activeStroke = null;
+
+  const [mostConfidentPrediction, ...lessConfidentAlternatives] = await drawing.getPrediction();
+  if (mostConfidentPrediction) {
+    console.log(mostConfidentPrediction.text);
```

Comment:
Anything else in there apart from `.text`?

Reply:
Should probably also log the `lessConfidentAlternatives`.

Reply:
Yes, the `segmentationResult` that is discussed in the following section.

Reply:
```js
  if (mostConfidentPrediction) {
    console.log(mostConfidentPrediction.text);
  }
  lessConfidentAlternatives?.forEach(alternative => console.log(alternative.text));
```

Reply:
Done. Is it okay to only mention `text` here?